python/lib/core/dmod/core/dataset.py

```diff
@@ -671,6 +686,41 @@ def filter(self, base_dataset: Dataset, restrictions: List[Union[ContinuousRestr
         """
         pass
 
+    def get_dataset_user(self, uuid: UUID) -> DatasetUser:
```

Comment:
The intent for the function is that it will only receive a user UUID value corresponding to a user instance that the manager is linked with (perhaps the caller even received the UUID from the manager; that's done in the _ServiceManager_ in one spot).  If some other UUID is given, it's fundamentally a problem.

I'd also expect real-world calls without a proper corresponding user object to generally be bug related: e.g., the manager failed to completely track a user properly when it was linked.   An immediate error made more sense to me there.